### ISSUE TITLE
Messages misidentification

### ISSUE THREAD

**Issue by apiote** [emersion/hydroxide#33]
After pulling from ProtonMail local mailbox is not the same as remote (that on PM servers).

Messages that were deleted in PM web client are still present in the local mailbox (others are deleted), on new mail in Inbox there are sometimes messages that were already read but are once again marked unread (from the same thread or other), archiving in web client actually archives different messages.

**Comment by emersion** [comment]
Yeah, the IMAP server part of hydroxide still needs a lot of testing and bugfixing.
